### PR TITLE
KTL comments on initial draft, not to be merged.

### PR DESCRIPTION
**Pull request by ktlim**

### CODE REVIEW

File: DMTN-110.tex [hunk 1]
```diff
@@ -23,7 +23,9 @@
 % \setDocCurator{The Curator of this Document}
 
 \setDocAbstract{%
-It is proposed  an improved way to deal with conda environment when building the science pipeline software.
+This proposal is an improved way to deal with Conda environments when building the Science Pipelines software.
+The benefits of implementing it can be seen in both development and release process.
```
Comment (ktlim) [inline DMTN-110.tex]: "development" → "the development"; "process" → "processes".

File: body.tex [hunk 2]
```diff
@@ -0,0 +1,417 @@
+\section{Introduction} \label{sec:intro}
+
+The proposal made in this document is oriented to have a more functional approach in the environment management for Science Pipelines builds.
+
+The benefits that can be obtained implementing this proposal are mainly for developers, who will have more control over the environments they are using.
```
Comment (ktlim) [inline body.tex]: "implementing" → "by implementing"

File: body.tex [hunk 3]
```diff
@@ -0,0 +1,417 @@
+\section{Introduction} \label{sec:intro}
+
+The proposal made in this document is oriented to have a more functional approach in the environment management for Science Pipelines builds.
+
+The benefits that can be obtained implementing this proposal are mainly for developers, who will have more control over the environments they are using. 
+This will permit a more dynamic evolution of the used environment, therefore more often upgrade to newer version of the packages.
```
Comment (ktlim) [inline body.tex]: "therefore more often upgrade" → "allowing more frequent upgrades"; "version" → "versions"; "the packages" → "packages in the environment"

File: body.tex [hunk 4]
```diff
@@ -0,0 +1,417 @@
+\section{Introduction} \label{sec:intro}
+
+The proposal made in this document is oriented to have a more functional approach in the environment management for Science Pipelines builds.
+
+The benefits that can be obtained implementing this proposal are mainly for developers, who will have more control over the environments they are using. 
+This will permit a more dynamic evolution of the used environment, therefore more often upgrade to newer version of the packages. 
+
+The release process will  benefit as the global build time for the Science Pipelines  will be reduced. 
+Reduced build times allows more frequent releases.
```
Comment (ktlim) [inline body.tex]: "Reduced build times" → "A reduced build time"; append "and faster development turnaround both locally and using continuous integration."

File: body.tex [hunk 5]
```diff
@@ -0,0 +1,417 @@
+\section{Introduction} \label{sec:intro}
+
+The proposal made in this document is oriented to have a more functional approach in the environment management for Science Pipelines builds.
+
+The benefits that can be obtained implementing this proposal are mainly for developers, who will have more control over the environments they are using. 
+This will permit a more dynamic evolution of the used environment, therefore more often upgrade to newer version of the packages. 
+
+The release process will  benefit as the global build time for the Science Pipelines  will be reduced. 
+Reduced build times allows more frequent releases. 
+This will be very important during commissioning and first operations phase, when multiple releases per day can be needed.
```
Comment (ktlim) [inline body.tex]: "first" → "the initial"

File: body.tex [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+
+In this moment there are a few \textit{stubs} as EUPS package that are not always working as expected.
```
Comment (ktlim) [inline body.tex]: "In this" → "At the"; "stubs as EUPS package" → "stub EUPS packages"

File: body.tex [hunk 7]
```diff
@@ -0,0 +1,417 @@
...
+Implementing this proposal will remove these packages and the overall build process will be more robust.
```
Comment (ktlim) [inline body.tex]: Comma before "and".

File: body.tex [hunk 8]
```diff
@@ -0,0 +1,417 @@
...
+
+The effort required to implement this proposal is mainly limited to a small team, that shall take ownership of the environment management activities.
```
Comment (ktlim) [inline body.tex]: Remove comma.

File: body.tex [hunk 9]
```diff
@@ -0,0 +1,442 @@
...
+
+\subsubsection{New envconfig Command} \label{sec:envconfig}
+
+A new command shall be added, that permits to choose which environment to activate.
+
+It shall accept one of following options:
+
+\begin{itemize}
+\item a \textit{scipipe\_conda\_env} environment reference to activate
+\item an existing build number, in order to activate the environment that has been used for that build.
```
Comment (ktlim) [inline body.tex:288]: You should make clear that this only works for already-deployed environments.

File: body.tex [hunk 10]
```diff
@@ -0,0 +1,442 @@
...
+The environment shall have a unique name, not the fixed one that is currently used.
+
+
+\subsubsection{Update rebuild Command} \label{sec:rebuild}
+
+The \textbf{rebuild} command should record the environment reference used for each build.
+
+This will permit in the future to retrieve the information, in order to manually activate the same conda environment before running a build.
+
+The \textbf{rebuild} command could also raise a warning, in case the active environment is different from the one previously used.
```
Comment (ktlim) [inline body.tex]: You should make clear what "previously used" means and that it may be tricky to determine.

File: body.tex [hunk 11]
```diff
@@ -0,0 +1,442 @@
...
+\subsection{Changes to Jenkins builds} \label{sec:cibuilds}
+
+The Jenkins jobs, for example \textit{run-rebuild}, shall be updated in order to accept an environment reference as an optional parameter.
+In this way it will be possible to verify that the science pipelines builds with a new environment.
+
+
+\newpage
+\section{Following activities} \label{sec:follows}
+
+The changes proposed in the previous section \ref{sec:proposed} can be implemented with very minimum impact to the day by day activities.
```
Comment (ktlim) [inline body.tex]: "very minimum" → "minimal" or "very little".  "the day by day" → "day to day".

File: body.tex [hunk 12]
```diff
@@ -0,0 +1,442 @@
...
+
+The new functionalities introduced, can facilitate already the management of the conda environment as it is done now, and the development activities in general.
```
Comment (ktlim) [inline body.tex]: Remove both commas.  "facilitate already" → "already facilitate".  Remove "the" before "development".

File: body.tex [hunk 13]
```diff
@@ -0,0 +1,442 @@
...
+
+
+\subsection{Conda Packaging 3rd Party Libraries} \label{sec:3rdpkgs}
+
+Some of the packages listed in appendix \ref{sec:tomove} are already available in Conda public channels or \textit{PyPi} (see following \ref{sec:pypi}).
+They are now included in the Science Pipelines stack as EUPS packages since when they were initially needed, they were not available in any Conda channels.
+
+Some other packages require to be updated more often, therefore they have been included as EUPS packages, since the Conda environment is not sufficiently dynamic.
+
+Packages that requires customization, need to remain part of the stack, at least in a first phase.
```
Comment (ktlim) [inline body.tex]: Make clear that "customization" here is packages with substantial forks of the upstream, not packages with minor patches that can be handled by a new recipe.

File: body.tex [hunk 14]
```diff
@@ -0,0 +1,442 @@
...
+In case a pip package will bring into the environment too many PyPi dependencies, DM can decide to maintain a Conda recipe for it.
+
+
+\subsection{Change Control}
+
+Since maintaining conda recipes may require a considerable amount of effort, the DMCCB shall monitor when new packages are added to the environment or are used in the codebase.
+
+Changes to the environment shall be proposed using RFC Jira issues.
+
+Third party packages shall also be censed and documented together with the DM product tree.
```
Comment (ktlim) [inline body.tex]: Replace "censed"; not sure what it is supposed to mean.

File: body.tex [hunk 15]
```diff
@@ -0,0 +1,442 @@
...
+
+
+\subsection{Change Control}
+
+Since maintaining conda recipes may require a considerable amount of effort, the DMCCB shall monitor when new packages are added to the environment or are used in the codebase.
+
+Changes to the environment shall be proposed using RFC Jira issues.
+
+Third party packages shall also be censed and documented together with the DM product tree. 
+Each package shall have an internal reference person.
```
Comment (ktlim) [inline body.tex:359]: "Responsible" instead of "reference"?

**Review by ktlim — COMMENTED** [review]
A few requests for more detailed clarification, plus some (but definitely not all) spelling and grammar fixes.

File: body.tex (hunk 6 above)
Comment (gcomoretto) [reply body.tex]: It doesn't read fluid, I rephrase.

File: body.tex (hunk 10 above)
Comment (gcomoretto) [reply body.tex]: I would say: "However this is not a problem related to the environment management but a potential improvement in the build tooling".
If these sentences are confusing... I will remove them. They are partially out of scope here.

File: body.tex (hunk 12 above)
Comment (gcomoretto) [reply body.tex]: I removed a bit more

File: body.tex (hunk 15 above)
Comment (gcomoretto) [reply body.tex:359]: "responsible" mean responsibilities... and I don't see people be very happy to assume responsibilities. Using the word "reference" do not imply responsibilities.

If DM Arch, or DMCCB is willing to enforce an internal responsible on each 3rd party package, I will be very happy, however, having a reference person for the time been should be sufficient.

**Pull request merged**